frontend/packages/kubevirt-plugin/src/selectors/vm/vmlike.ts

```diff
@@ -0,0 +1,22 @@
+import { K8sKind, TemplateKind } from '@console/internal/module/k8s';
+import { TemplateModel } from '@console/internal/models';
+import { VMLikeEntityKind, VMKind } from '../../types';
+import { VirtualMachineModel } from '../../models';
+import { selectVM } from '../vm-template/selectors';
+
+export const isVM = (vmLikeEntity: VMLikeEntityKind): boolean =>
+  vmLikeEntity && vmLikeEntity.kind === VirtualMachineModel.kind;
+
+export const getVMLikeModel = (vmLikeEntity: VMLikeEntityKind): K8sKind =>
+  isVM(vmLikeEntity) ? VirtualMachineModel : TemplateModel;
+
+export const asVM = (vmLikeEntity: VMLikeEntityKind): VMKind => {
+  if (!vmLikeEntity) {
+    return null;
+  }
+
+  if (isVM(vmLikeEntity)) {
```

Comment:
when the above is converted to type guard you can just

```suggestion
  return isVM(vmLikeEntity) ? vmLikeEntity : selectVM(vmLikeEntity);
```

Reply:
done